### PR TITLE
Add `is.class()`

### PR DESCRIPTION
**Pull request by SchneiderOr**

### CODE REVIEW

**Comment by sindresorhus** [comment]
When would you need to know if something is a `class`? I'm genuinely curious. I considered adding this initially, but couldn't think of when I would use it,

**Comment by sudo-suhas** [comment]
Maybe to see if `new` was required for invoking?

**Comment by SchneiderOr** [comment]
In my use case i have a tool that parsing react components to extract their displayName, propTypes, defaultProps etc... based on some strategies, one of them is to check wether the component you intent to extract props from is an ES6 class instance or a functional component and then handle the parsing and extracting of the data differently.

File: test.js [hunk 1]
```diff
@@ -130,6 +134,10 @@ test('is.array', t => {
 	testType(t, 'array');
 });
 
+test('is.class', t => {
+	testType(t, 'function');
```
Comment (sindresorhus) [inline test.js]: `function` => `class`

File: index.js [hunk 2]
```diff
@@ -62,6 +62,7 @@ is.symbol = x => typeof x === 'symbol';
 
 is.array = Array.isArray;
 is.function = x => typeof x === 'function';
+is.class = x => typeof x === 'function' && /^\s*class\s+/.test(x.toString());
```
Comment (sindresorhus) [inline index.js]: `/^\s*class\s+/.test(x.toString());` => `x.toString().startsWith('class ');`

I don't think `.toString()` includes leading whitespace.

**Comment by sindresorhus** [comment]
Also needs to be added to the readme.

**Comment by SchneiderOr** [comment]
Okey, tests fixed and i've added description to the readme.

**Comment by sindresorhus** [comment]
Thank you :)

**Comment by SchneiderOr** [comment]
Sure, nice package :)

**Comment by sudo-suhas** [comment]
A small note, this does not work if the class declaration was transpiled with babel.

**Pull request merged**